### DISCUSSION TITLE
Move item creation / editing to the drawer

### DISCUSSION THREAD

**Discussion by oreilles** [directus/directus#5120] [Feature Requests]
Currently, when clicking on an item from a layout, we are routed to the item page `/collections/primary_key`. 
When done editing, the default behaviour is to go back to the origin collection.

This poses two problems:
* All the collection items have to be re-fetched and the whole layout has to be redrawn: waste of time and resource.
* There is a loss of focus, as the whole page changes as well as the scroll position.

A better behaviour would be to default opening the item edit page in a drawer, and to only update that item back in the layout data. It would be particularly useful for the map layout as the conversion to geojson as well as the initial map load and render might be quite expansive. Here's a simple demo I made by creating a private component `drawer-item-edit` scraping pretty much all the code from `collection/item.vue` and replacing the routing by some events emitting.

https://user-images.githubusercontent.com/33065839/115112444-e5373780-9f85-11eb-82e8-f5c566d7cfc1.mp4

**Comment by benhaynes** [comment]
Love it! We've discussed this internally in the past, and I think it's definitely something we want to implement. The only question is where this is configured (open as page, vs open in drawer). Some options with different scopes:

* Project Settings (Global)
* Collection Settings
* Role Permissions
* User Preferences (Layout Options)

We could do multiple too, but that might be overkill. Personally, I think Collection as the default... and _maybe_ a User override in the layout sidebar options.

Thoughts? @rijkvanzanten @Oreilles

**Reply by oreilles** [reply]
It's already the default for batch editing, so wouldn't it make sense to also make it the default for single item editing ? 🙂 With a "open as page" link at the top ?

**Reply by benhaynes** [reply]
There was a technical reason for the batch edit being in the drawer: we were passing the limit of characters in the URL when having a CSV of _many_ UUIDs (drawer doesn't change the URL). There are many reasons that a user may prefer a dedicated page to a drawer:

* No URL change for drawers (as of now)
* Drawers don't have the right info sidebar (as of now)
* Lose quick access (one click) to navigation and modules
* Smaller editing area, also means `fill` size doesn't do anything

We can solve some of the issues above, but I see the drawer as a quicker "preview" and still think there should be an option for the user to choose how items open per Collection. Yes, the drawer should have a "Launch" icon to go to the dedicated page, but I don't want to force users to take that extra step if they don't want to.

**Reply by rijkvanzanten** [reply]
I think this is a per-layout kinda thing. It makes sense to me that the map layout would have a small popup preview "on" the marker on the map on click/hover with a CTA to open the "full page", while a calendar might rely on a dialog for the event "preview", while the cards layout might navigate directly to the full page 🤔

**Reply by benhaynes** [reply]
Maybe the Layout can set a default or override, but I do think it's useful to give control to the user for the Table Layout. I suppose each layout could control its own, and Table could have an ~~interface~~ layout option for choosing.

**Reply by oreilles** [reply]
On my laptop, the full page item editing doesn't offer more space than the drawer because `--form-column-max-width: 380px`, so a full width interface couldn't be larger than 760px, but `--v-drawer-max-width: 856px;`. 🙂

If it had to be a setting, I think it would make sense to make it a per-group per-layout setting. I don't think it's the end user role to change that behaviour.

**Reply by benhaynes** [reply]
I think you're confusing `full` for `fill`... those are two separate widths, and the latter will expand beyond the form column max width. 
![Screen Shot 2021-04-19 at 2 06 13 PM](https://user-images.githubusercontent.com/522079/115282680-79d19f00-a118-11eb-8573-c23edb25baaa.png)

**Reply by oreilles** [reply]
Oh wow, I wasn't even aware this was a setting.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
